### PR TITLE
Remove redundant cmp_owned and comments

### PR DESCRIPTION
**Pull request by teor2345**
## Motivation

This is a cleanup after PR #2184, where we forgot to:
- delete some comments resolved by the PR, and
- re-enable some lints resolved by the PR.

## Review

@dconnolly or @oxarbitrage can review this, since they did the original PR.

### CODE REVIEW

File: zebra-chain/src/orchard/keys.rs [hunk 1]
```diff
@@ -315,17 +313,13 @@ impl From<SpendAuthorizingKey> for SpendValidatingKey {
 }
 
 impl PartialEq for SpendValidatingKey {
-    #[allow(clippy::cmp_owned)]
     fn eq(&self, other: &Self) -> bool {
-        // XXX: These redpallas::VerificationKey(Bytes) fields are pub(crate)
         self.0.bytes.bytes == other.0.bytes.bytes
```
Comment (teor2345) [inline zebra-chain/src/orchard/keys.rs:318]: Keep this change
```suggestion
        // XXX: These redpallas::VerificationKey(Bytes) fields are pub(crate)
        self.0.bytes.bytes == other.0.bytes.bytes
```

File: zebra-chain/src/orchard/keys.rs [hunk 2]
```diff
@@ -315,17 +313,13 @@ impl From<SpendAuthorizingKey> for SpendValidatingKey {
...
     }
 }
 
 impl PartialEq<[u8; 32]> for SpendValidatingKey {
-    #[allow(clippy::cmp_owned)]
     fn eq(&self, other: &[u8; 32]) -> bool {
-        // XXX: These redpallas::VerificationKey(Bytes) fields are pub(crate)
         self.0.bytes.bytes == *other
```
Comment (teor2345) [inline zebra-chain/src/orchard/keys.rs:325]: Keep this change
```suggestion
        // XXX: These redpallas::VerificationKey(Bytes) fields are pub(crate)
        self.0.bytes.bytes == *other
```

**Review by teor2345 — COMMENTED** [review]
I think we need to keep these comments that were added in the previous PR

**Review by dconnolly — APPROVED** [review]
Ah thank you for catching these 🤩

**Pull request merged**